otherlibs/unix/link.c

```diff
@@ -18,18 +18,32 @@
 #include <caml/signals.h>
 #include "unixsupport.h"
 
-CAMLprim value unix_link(value path1, value path2)
+#include <fcntl.h>
+#include <unistd.h>
+
+CAMLprim value unix_link(value follow, value path1, value path2)
 {
-  CAMLparam2(path1, path2);
+  CAMLparam3(follow, path1, path2);
   char * p1;
   char * p2;
   int ret;
   caml_unix_check_path(path1, "link");
   caml_unix_check_path(path2, "link");
   p1 = caml_strdup(String_val(path1));
   p2 = caml_strdup(String_val(path2));
+  ret =
+    Is_block(follow) && Bool_val(Field(follow, 0)) /* Some true */
+    ? AT_SYMLINK_FOLLOW
+    : 0;
   caml_enter_blocking_section();
-  ret = link(p1, p2);
+  if (follow == Val_int(0) /* None */)
+    ret = link(p1, p2);
+  else /* Some bool */
+# if _XOPEN_VERSION >= 700 || _POSIX_VERSION >= 200809L
```

Comment:
Indeed - I'm not certain how this is supposed to work with glibc, as you do not appear to be permitted to check the version of the glibc headers you are compiling. My (quite possibly incorrect) understanding is that if someone is using glibc < 2.10 then they should ensure that -D_ATFILE_SOURCE gets into CFLAGS at configure-time (it's possible that this is something configure itself should determine is required, that's not clear at all).

However, it does seem to me to be OK to test it here.